### PR TITLE
Adding contributing guidelines

### PR DESCRIPTION
**Pull request by Healarious**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,33 @@ For some examples of what WeakAuras can do, take a look at [wago.io](https://wag
 * You're a programmer yourself and want to contribute? We accept [pull requests](https://github.com/WeakAuras/WeakAuras2/pulls) on GitHub!
 * Feel free to join our [Discord Community](https://discord.gg/wa2) at https://discord.gg/wa2 to talk and discuss everything WeakAuras!
 
+## Contributing to WeakAuras 2
+* Fork the WeakAuras 2 repository
+* Clone your fork to your local disk
+* Create a new branch, `Somebug-patch` for example and move into it
+  ```
+  $ git branch Somebug-patch
+  $ git checkout Somebug-patch
```
Comment (InfusOnWoW) [inline README.md]: git checkout -b somebug-patch

File: README.md [hunk 2]
```diff
@@ -35,6 +35,33 @@ For some examples of what WeakAuras can do, take a look at [wago.io](https://wag
...
+### Keeping your fork updated
+* Specify a new remote upstream repository that will be used to sync your fork (you only need to do this the first time around)
+  ```
+  git remote add upstream https://github.com/WeakAuras/WeakAuras2.git
+  ```
+* In order to sync the fork with the original WeakAuras 2 repository you would do
+  ```
+  $ git fetch upstream
+  $ git checkout master
+  $ git merge upstream/master
```
Comment (InfusOnWoW) [inline README.md]: git rebase upstream/master (that works better if the user has local changes.)

**Comment by krazyito65** [comment]
Probably should go on the wiki, or with more detail

**Comment by Stanzilla** [comment]
Hmm not sure, I actually like it in the readme, the other common thing projects do is having a `CONTRIBUTING.MD` file like for ex Atom has https://github.com/atom/atom/blob/master/CONTRIBUTING.md

**Comment by Stanzilla** [comment]
Also might be worth merging with some parts of https://github.com/WeakAuras/WeakAuras2/wiki/Developing-WeakAuras

**Comment by krazyito65** [comment]
That looks good as well. Main thing I was thinking of was just to have a place we could add more details. Linking it via the readme

**Comment by Healarious** [comment]
I indeed kept it rather short because I did not want the readme to become rather large

**Comment by Stanzilla** [comment]
This is pretty neat https://github.com/blog/1184-contributing-guidelines, so we should go that route, just still wondering what exactly should be in there, since most of it (this PR and the Dev WA page) is not really WeakAuras related but more general developing and Lua.

**Comment by Healarious** [comment]
Well I agree but general developing (Lua/git style) is also part of developing weakauras 2, some ppl (like me for example) did not already know those things. Maybe just adding a very brief summary of those things and just point to other resources (Github's wiki itself) for more in-depth explanations, see it like a small section to quickly refresh on those things

**Comment by Healarious** [comment]
Going to edit the file all up today (sometime today)

**Comment by Stanzilla** [comment]
can you remove the readme.md change, please? I'll probably merge and add a few things from https://github.com/XhmikosR/mpc-hc/blob/develop/CONTRIBUTING.md that I find useful afterwards

**Pull request closed without merging**